cloud-infrastructure/storage/block-storage/asset/trim-showcase.md

```diff
@@ -0,0 +1,199 @@
+sudo mkdir /mnt/MyBlockVolume
+sudo umount ${DATAoDIR}
+sudo mkfs -t ext4 ${DEVICEoPATH}
+
+sudo mount ${DEVICEoPATH} ${DATAoDIR}
+sudo chown -R opc:opc ${DATAoDIR}
+ls -lah ${DATAoDIR}
+sudo mount | grep ${DATAoDIR}
+df -h | grep ${DATAoDIR}
+sudo fstrim --all
+~~~
+
+### This will run ~ 20 minutes
+~~~
+echo "Start Trim Test (trim NOT active)" > "${myLOGFILE}"
+echo "============================================================================================" >> "${myLOGFILE}"
+for i in $( seq 1 $myITERATIONS )
+do
+
+  if [ 1 -eq 1 ] ; then # show Size Information: Filesystem
+    echo -n "Size Information: Filesystem..: " >> "${myLOGFILE}"
+    df -h | grep /mnt/MyBlockVolume >> "${myLOGFILE}"
+  fi
+
+  if [ 1 -eq 1 ] ; then # take backup, show Size Information: Block Volume; delete backup
+    oci --profile "${PROFILE_FRANKFURT}" bv backup create --volume-id "${TRIM_TEST_OCID}" --display-name "${TRIM_TEST_BACKUP_NAME}" --wait-for-state "AVAILABLE"
+
+    oci --profile "${PROFILE_FRANKFURT}" bv backup list --compartment-id "${COMPARTMENT_OCID}" --display-name "${TRIM_TEST_BACKUP_NAME}" > "${myTEMPFILE}"
+    for ii in $(cat "${myTEMPFILE}"|jq --raw-output '.[] | [.[] | select(."display-name"=="TrimTestBackup")] | .[]."id" ')
+    do
+      TRIM_TEST_BACKUP_OCID=$ii
+      TRIM_TEST_BACKUP_SIZE_IN_MB=$( cat "${myTEMPFILE}"|jq --raw-output '.[] | [.[] | select(.'\"id\"'=='\"$ii\"')] | .[].'\"unique-size-in-mbs\"' ' )
+    done
+    echo "Size Information: Block Volume: Backup size in MB: ${TRIM_TEST_BACKUP_SIZE_IN_MB}" >> "${myLOGFILE}"
+
+    echo "Action..........: Block Volume: delete backup" >> "${myLOGFILE}"
+    oci --profile "${PROFILE_FRANKFURT}" bv backup delete --volume-backup-id "${TRIM_TEST_BACKUP_OCID}" --force --wait-for-state "TERMINATED"
+  fi
+
+  if [ 1 -eq 1 ] ; then # insert, update delete files
+    echo "Action..........: Filesystem..: insert, update and delete" >> "${myLOGFILE}"
+    dd if=/dev/zero of=/mnt/MyBlockVolume/iteration.$i.file1.txt count=$mySIZE bs=1024
+    dd if=/dev/zero of=/mnt/MyBlockVolume/iteration.$i.file2.txt count=$mySIZE bs=1024
+    dd if=/dev/zero of=/mnt/MyBlockVolume/iteration.$i.file1.txt count=$mySIZE bs=1024
+    rm -f /mnt/MyBlockVolume/iteration.$i.file1.txt
+	#sudo fstrim --all
+  fi  
+
+done
+
+cat "${myLOGFILE}"
+~~~
+
+## TRIM ON showcase
+
+### Set needed variables (*please adjust variables to your needs before past into your terminal*)
+~~~
+# ---------------------------------------------------------------------------------------------------------------------------------------------
+# CUSTOMER SPECIFIC VALUES - please update appropriate
+# ---------------------------------------------------------------------------------------------------------------------------------------------
+export TENENCY_OCID=<your_tenency_ocid>
+export USER_OCID=<your_user_ocid>
+export COMPARTMENT_OCID=<your_compartment_ocid>
+export FRANKFURT_SUBNET_OCID=<your_frankfurt_subnet_ocid>
```

Comment:
Again why FRANKFURT? Can this be region neutral?

Reply:
done